api/k8s/v1beta4/k8s_helpers.go

```diff
@@ -47,3 +47,69 @@ func (s *API) WaitForCluster(req *WaitForClusterRequest) (*Cluster, error) {
 	}
 	return cluster.(*Cluster), nil
 }
+
+// WaitForClusterNodesReadyRequest is used by WaitForClusterNodesReady method.
+type WaitForClusterNodesReadyRequest struct {
+	ClusterID string
+	Region    scw.Region
+	Timeout   time.Duration
+}
+
+// WaitForClusterNodesReady waits for the nodes of a cluster to be ready
+func (s *API) WaitForClusterNodesReady(req *WaitForClusterNodesReadyRequest) error {
+	pools, err := s.ListPools(&ListPoolsRequest{
+		ClusterID: req.ClusterID,
+		Region:    req.Region,
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, pool := range pools.Pools {
+		err = s.WaitForPoolNodesReady(&WaitForPoolNodesReadyRequest{
+			PoolID:  pool.ID,
+			Timeout: req.Timeout,
+		})
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// WaitForPoolNodesReadyRequest is used by WaitForPoolNodesReady method.
+type WaitForPoolNodesReadyRequest struct {
+	PoolID  string
+	Region  scw.Region
+	Timeout time.Duration
+}
+
+// WaitForPoolNodesReady waits for the nodes of a pool to be ready
+func (s *API) WaitForPoolNodesReady(req *WaitForPoolNodesReadyRequest) error {
+	if req.Region == "" {
```

Comment:
Okay sorry my bad, but it's already implemented in the GetPool 😅

Reply:
I saw it was in ListPools but didn't check in GetPool. Removing